component/src/main/java/com/synopsys/integration/alert/component/settings/encryption/web/SettingsEncryptionController.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * component
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.alert.component.settings.encryption.web;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import com.synopsys.integration.alert.common.rest.ResponseFactory;
+import com.synopsys.integration.alert.common.rest.api.BaseResourceController;
+import com.synopsys.integration.alert.common.rest.api.ValidateController;
+import com.synopsys.integration.alert.common.rest.model.ValidationResponseModel;
+import com.synopsys.integration.alert.component.settings.encryption.action.SettingsEncryptionCrudActions;
+import com.synopsys.integration.alert.component.settings.encryption.action.SettingsEncryptionValidationAction;
+import com.synopsys.integration.alert.component.settings.encryption.model.SettingsEncryptionModel;
+
+//TODO: Disabled until new UI component is complete
+//@RestController
+//@RequestMapping(AlertRestConstants.SETTINGS_ENCRYPTION_PATH)
+public class SettingsEncryptionController implements BaseResourceController<SettingsEncryptionModel>, ValidateController<SettingsEncryptionModel> {
```

Comment:
A bit of a heads up, depending on how the UI is implemented/will be implemented you might need a get page endpoint. If you use the getOne endpoint, where will you get the original ID from? Alternatively, we could come up with a new type of Controller that assumes it will only ever have a single row that it always needs to return/modify.

Reply:
In my discussions with Paulo and James we only ever plan to have a single encryption configuration. Within the SettingsEncryptionCrudActions we have a DEFAULT_ENCRYPTION_ID as a static ID. Our plan was for the UI to only ever call that same ID. Likewise, we have set validation within  the getSettingsEncryptionModel should the ID not match the static encryption ID.

Reply:
That said, I do think if we had a new type of controller that only ever expected a single value to return or modify, then it would simplify this somewhat and remove that ambiguity.